### PR TITLE
feat(components): [image-viewer] Add rotate event when rotating images

### PR DESCRIPTION
**Pull request by HADB**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at 2920ecd</samp>

This pull request adds a new `rotate` event to the `Image Viewer` component and its documentation. The event allows users to listen for changes in the image rotation and get the current rotation degree.

## Related Issue

Fixes #14136.

### <samp>🤖 Generated by Copilot at 2920ecd</samp>

*  Add a new event `rotate` to the `Image Viewer` component that is triggered when the user rotates the image and passes the current rotation degree as a parameter ([link](https://github.com/element-plus/element-plus/pull/14138/files?diff=unified&w=0#diff-b54969024ba1fc4abdee254a61d8c974d4ec76252e1ebbccebc2b8181a6ab14dR119), [link](https://github.com/element-plus/element-plus/pull/14138/files?diff=unified&w=0#diff-ba7704f324b9e09836ac7ce64c03e1ab15cd30f786ebd706e70f1bfe6b800d20R73), [link](https://github.com/element-plus/element-plus/pull/14138/files?diff=unified&w=0#diff-0e01148bcd3b0455959cc7582dac606121762fab15bce2e6976225e9faac2f1eL348-R352))
  * Update the documentation of the component in `docs/en-US/component/image.md` to include the new event and its parameter ([link](https://github.com/element-plus/element-plus/pull/14138/files?diff=unified&w=0#diff-b54969024ba1fc4abdee254a61d8c974d4ec76252e1ebbccebc2b8181a6ab14dR119))
  * Add a validator function for the event parameter in `packages/components/image-viewer/src/image-viewer.ts` to ensure it is a number ([link](https://github.com/element-plus/element-plus/pull/14138/files?diff=unified&w=0#diff-ba7704f324b9e09836ac7ce64c03e1ab15cd30f786ebd706e70f1bfe6b800d20R73))
  * Emit the event from the component in `packages/components/image-viewer/src/image-viewer.vue` whenever the user clicks the clockwise or anticlockwise buttons and pass the updated rotation degree as a parameter ([link](https://github.com/element-plus/element-plus/pull/14138/files?diff=unified&w=0#diff-0e01148bcd3b0455959cc7582dac606121762fab15bce2e6976225e9faac2f1eL348-R352))

### CODE REVIEW

**Comment by HADB** [comment]
@btea Hi, could you please take a look at this PR when you have time? Any chance of getting it merged?

**Comment by btea** [comment]
I don't think it would be useful to emit this event, can you elaborate on your rationale?

**Comment by btea** [comment]
Based on your description, I think using the `image` component for image composition may not be very flexible. Currently `image-viewer` rotates the image every step 90 degrees, it may not be very convenient for some fine angle adjustments.

According to your actual scenario, I think using a library similar to [fabric.js](https://github.com/fabricjs/fabric.js) may be more appropriate.🤔

**Comment by HADB** [comment]
> Based on your description, I think using the `image` component for image composition may not be very flexible. Currently `image-viewer` rotates the image every step 90 degrees, it may not be very convenient for some fine angle adjustments.
> 
> 
> 
> According to your actual scenario, I think using a library similar to [fabric.js](https://github.com/fabricjs/fabric.js) may be more appropriate.🤔

Fabric.js is too heavy for this scenario because I just need the direction of rotation like "clockwise" or "anticlockwise". Or is it more acceptable to emit rotate "clockwise/anticlockwise" instead of the degrees?

**Review by btea — APPROVED** [review]

**Review by HeftyKoo — APPROVED** [review]

File: docs/en-US/component/image.md [hunk 1]
```diff
@@ -116,6 +116,7 @@ image/image-preview
 | ------ | ------------------------------------------------------------------------------------------------- | ------------------------------------ |
 | close  | trigger when clicking on close button or when `hide-on-click-modal` enabled clicking on backdrop. | ^[Function]`() => void`              |
 | switch | trigger when switching images.                                                                    | ^[Function]`(index: number) => void` |
+| rotate | trigger when rotating images.                                                                     | ^[Function]`(deg: number) => void`   |
```
Comment (btea) [inline docs/en-US/component/image.md:119]:
```suggestion
| rotate ^(2.3.13) | trigger when rotating images.                                                                     | ^[Function]`(deg: number) => void`   |
```

**Pull request merged**